### PR TITLE
util/chunk: some minor refactors

### PR DESCRIPTION
**Pull request by xxchan**
### What problem does this PR solve?
splitted from https://github.com/pingcap/tidb/pull/26772

Problem Summary:

some minor refactors
- add some comments
- remove unused `preAlloc` and `insert`
- remove unused parameter `old` in `newVarLenColumn(cap int, old *Column)`

### Release note 

```release-note
None
```

### CODE REVIEW

File: util/chunk/chunk.go [hunk 1]
```diff
@@ -394,7 +371,8 @@ func (c *Chunk) AppendRowByColIdxs(row Row, colIdxs []int) (wide int) {
 	return
 }
 
-// AppendPartialRowByColIdxs appends a row by its colIdxs to the chunk.
+// AppendPartialRowByColIdxs appends a row to the chunk starting from colOff,
+// using the row‘s columns specified by colIdxs.
```
Comment (xiongjiwei) [inline util/chunk/chunk.go:375]: looks like it is a Chinese single quote 😄
```suggestion
// using the row's columns specified by colIdxs.
```

**Comment by xiongjiwei** [comment]
there is a test case failed, I will update the branch and re-run it.

**Comment by xxchan** [comment]
/cc @XuHuaiyu @lysu 

PreAlloc4Row and Insert were introduced in https://github.com/pingcap/tidb/pull/7916. Can they be removed?

**Review by xhebox — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**